src/flowcean/torch/lightning_learner.py

```diff
@@ -127,3 +131,261 @@ def configure_optimizers(self) -> Any:
+
+    def __init__(
+        self,
+        learning_rate: float,
+        conv_configs: list[tuple[int, int, int]] | None = None,
+        fully_connected_layer_sizes: list[int] | None = None,
+        output_size: int = 10,
+        dropout_rates: list[float] | None = None,
+        input_shape: tuple[int, int, int] = (1, 28, 28),
+    ) -> None:
+        """Initialize the model.
+
+        Args:
+            learning_rate: The learning rate.
+            conv_configs: The convolutional layer configurations.
+            fully_connected_layer_sizes: The fully connected layer sizes.
+            output_size: The size of the output.
+            dropout_rates: The dropout rates.
+            input_shape: The shape of the input.
+        """
+        super().__init__()
+        if conv_configs is None:
+            conv_configs = [(1, 32, 3)]
+        if fully_connected_layer_sizes is None:
+            fully_connected_layer_sizes = [128]
+        if dropout_rates is None:
+            dropout_rates = [0.25, 0.5]
+        self.input_shape = input_shape
+        self.save_hyperparameters()
+        self.learning_rate = learning_rate
+
+        # Convolutional layers
+        self.conv_layers = torch.nn.ModuleList()
+        for in_ch, out_ch, kernel_size in conv_configs:
+            self.conv_layers.append(
+                torch.nn.Conv2d(in_ch, out_ch, kernel_size=kernel_size),
+            )
+
+        # Fully connected layers
+        self.fully_connected_layers = torch.nn.ModuleList()
+        prev_size = self._compute_flat_features(conv_configs)
+        for size in fully_connected_layer_sizes:
+            self.fully_connected_layers.append(
+                torch.nn.Linear(prev_size, size),
+            )
+            prev_size = size
+        self.output_layer = torch.nn.Linear(prev_size, output_size)
+
+    def _compute_flat_features(
+        self,
+        conv_configs: list[tuple[int, int, int]],
+    ) -> int:
+        channels, height, width = self.input_shape
+        for _, out_ch, kernel_size in conv_configs:
+            height = height - kernel_size + 1  # Conv without padding
+            width = width - kernel_size + 1
+            height //= 2  # MaxPool2d with kernel 2
+            width //= 2
+            channels = out_ch
+        return channels * height * width
+
+    def forward(self, x: Tensor) -> Tensor:
+        for conv in self.conv_layers:
+            x = torch.relu(conv(x))
+            x = torch.max_pool2d(x, 2)
+        x = torch.flatten(x, 1)
+        for fc in self.fully_connected_layers:
+            x = torch.relu(fc(x))
+        logits = self.output_layer(x)
+        probabilities = torch.sigmoid(logits)
+        return torch.round(probabilities)
+
+    def training_step(self, batch: Any) -> Tensor:
+        inputs, targets = batch
+        outputs = self(inputs)
+        targets = targets.float()
+        return torch.nn.functional.binary_cross_entropy(outputs, targets)
```

Comment:
this is hard for the user to see. Why is this a BCE loss, and the others are MSE?